bookwyrm/views/books/series.py

```diff
@@ -0,0 +1,66 @@
+""" books belonging to the same series """
+from django.views import View
+from django.shortcuts import get_object_or_404
+from django.template.response import TemplateResponse
+
+from bookwyrm.views.helpers import is_api_request
+from bookwyrm import models
+
+
+# pylint: disable=no-self-use
+class BookSeriesBy(View):
+    """book series by author"""
+
+    def get(self, request, author_id):
+        """lists all books in a series"""
+        series_name = request.GET.get("series_name")
+
+        if is_api_request(request):
+            pass
+
+        author = get_object_or_404(models.Author, id=author_id)
+
+        results = models.Edition.objects.filter(authors=author, series=series_name)
+
+        # when there are multiple editions of the same work
+        # pick the one with a series number or closest
+        work_ids = results.values_list("parent_work__id", flat=True).distinct()
+
+        # filter out multiple editions of the same work
+        numbered_books = []
+        dated_books = []
+        unsortable_books = []
+        for work_id in set(work_ids):
+            result = (
+                results.filter(parent_work=work_id)
+                .order_by("series_number", "-edition_rank")
+                .first()
+            )
+            if result.series_number:
+                numbered_books.append(result)
+            elif result.first_published_date or result.published_date:
+                dated_books.append(result)
+            else:
+                unsortable_books.append(result)
+
+        list_results = (
+            sorted(numbered_books, key=lambda book: float(book.series_number))
```

Comment:
I really want to approve this PR, but we have a bit of a problem here because `book.series_number` is a `CharField`, when it arguably should be an `IntegerField` or maybe a `DecimalField`. This means the form input will take any string, and if anyone makes the series number something like "Three" or "Book 6" with this sort it will throw a `ValueError` because it can't be coerced into a float.

This can be solved, but you'll have to give up on using a `lambda` and use named function instead so we can handle the exception. e.g. you could do something like this:

```python
class BookSeriesBy(View):

...

     list_results = (
            sorted(numbered_books, key=self.sort_by_series)
            + sorted(
                dated_book
...

    def sort_by_series(self, book):
        try:
            return float(book.series_number)
        except ValueError:
            return 999999999

```

Reply:
Would using `sys.float_info.max` make more sense instead of a magic number?

Reply:
Yes! 😄

Reply:
That should be fixed now as well 😊